src/sku/molecule/services/src/use_cases/molecule_enable_project_use_case_impl.rs

```diff
@@ -0,0 +1,154 @@
+    outbox: Arc<dyn Outbox>,
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[common_macros::method_names_consts]
+#[async_trait::async_trait]
+impl MoleculeEnableProjectUseCase for MoleculeEnableProjectUseCaseImpl {
+    #[tracing::instrument(
+        level = "info",
+        name = MoleculeEnableProjectUseCaseImpl_execute,
+        skip_all,
+        fields(?ipnft_uid)
+    )]
+    async fn execute(
+        &self,
+        molecule_subject: &LoggedAccount,
+        ipnft_uid: String,
+    ) -> Result<MoleculeProjectEntity, MoleculeEnableProjectError> {
+        use datafusion::prelude::*;
+
+        let (projects_dataset, df_opt) = self
+            .molecule_dataset_service
+            .get_projects_raw_ledger_data_frame(molecule_subject, DatasetAction::Write, false)
+            .await?;
+
+        let Some(ledger_df): Option<odf::utils::data::DataFrameExt> = df_opt else {
+            return Err(MoleculeEnableProjectError::ProjectNotFound(
+                ProjectNotFoundError {
+                    ipnft_uid: ipnft_uid.clone(),
+                },
+            ));
+        };
+
+        // Reconstruct active snapshot to confirm there are no conflicts
+        let snapshot_df = odf::utils::data::changelog::project(
+            ledger_df.clone(),
+            &["account_id".to_string()],
+            &odf::metadata::DatasetVocabulary::default(),
+        )
+        .int_err()?;
+
+        let df = ledger_df
+            .filter(col("ipnft_uid").eq(lit(&ipnft_uid)))
+            .int_err()?
+            .sort(vec![col("system_time").sort(false, false)])
+            .int_err()?;
+
+        let records: Vec<serde_json::Value> = df.collect_json_aos().await.int_err()?;
+        let record = records.into_iter().find(|record: &serde_json::Value| {
+            record
+                .get("op")
+                .and_then(serde_json::Value::as_u64)
+                .and_then(|value| u8::try_from(value).ok())
+                .and_then(|value| OperationType::try_from(value).ok())
+                .map(|op| matches!(op, OperationType::Append | OperationType::CorrectTo))
+                .unwrap_or(false)
+        });
+
+        let Some(record) = record else {
+            return Err(MoleculeEnableProjectError::ProjectNotFound(
+                ProjectNotFoundError {
+                    ipnft_uid: ipnft_uid.clone(),
+                },
+            ));
+        };
+
+        let mut project = MoleculeProjectEntity::from_json(record).int_err()?;
+        project.ipnft_symbol.make_ascii_lowercase();
+
+        let conflict_df = snapshot_df
+            .filter(
+                col("ipnft_uid")
+                    .eq(lit(&project.ipnft_uid))
+                    .or(lower(col("ipnft_symbol")).eq(lit(&project.ipnft_symbol))),
+            )
+            .int_err()?;
+
+        let conflicts = conflict_df.collect_json_aos().await.int_err()?;
```

Comment:
Here we are making another expensive query via datafusion which seems unnecessary. If we get the last entry of a changelog (ledger) with `ipnft_uid = X` - this should already tell us:
- whether project is in disabled state and can be enabled
- whether project is already in enabled state

Also not quite sure why you are referring to this as `conflict` - IPNFT_UID is a unique identified for projects, no matter if they are enabled or disabled, so it's not a conflict as such.


BTW: Please add test that when creating a project with symbol `X` and there is already project `X` but in disabled state - we still get a conflict error.